circuits/circom/membership.circom

```diff
@@ -0,0 +1,61 @@
+pragma circom 2.0.2;
+
+include "node_modules/circom-ecdsa/circuits/zk-identity/eth.circom";
+include "node_modules/circom-ecdsa/circuits/ecdsa.circom";
+include "./merkleTree.circom";
+
+// Proves that a message is signed by one of the addresses in a merkle tree
+// Essentially acts as a group signature for secp256k1
+// k is the number of registers and n is their bit length
+template InAddressSet(n, k, levels) {
```

Comment:
This is the core functionality for set membership. Note that circom-ecdsa requires big int arithmetic, since the secp256k1 curve is larger than the scalar field we have access to in standard circom.

This template essentially just calls other components that do the heavy lifting.